Sources/KituraNet/HTTP/HTTPServerRequest.swift

```diff
@@ -126,7 +126,16 @@ public class HTTPServerRequest: ServerRequest {
 
     private var enableSSL: Bool = false
 
-    private var _urlString: String
+    private var rawURLString: String
+
+    private var urlStringPercentEncodingRemoved: String?
+
+    private var _urlString: String {
+        if urlStringPercentEncodingRemoved == nil {
+            urlStringPercentEncodingRemoved = rawURLString.removingPercentEncoding ?? rawURLString
+        }
+        return urlStringPercentEncodingRemoved!
```

Comment:
I can see this code is correct, but could we use guard let here so that we don't need the force-unwrap?

Reply:
@djones6 Using `guard-let` and not force-unwrapping it can look only as good as this:
```
    private var _urlString: String {
        guard let urlStringPercentEncodingRemoved == self.urlStringPercentEncodingRemoved else {
            urlStringPercentEncodingRemoved = rawURLString.removingPercentEncoding ?? rawURLString
            return rawURLString.removingPercentEncoding ?? rawURLString
        }   
        return urlStringPercentEncodingRemoved
    }
```

Reply:
```swift
            let result = rawURLString.removingPercentEncoding ?? rawURLString
            urlStringPercentEncodingRemoved = result
            return result
```
It's important not to run `removingPercentEncoding` twice as it is quite expensive.

Reply:
Done. Thanks @djones6 !